x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authc/jwt/JwtRealmSettings.java

```diff
@@ -77,6 +78,37 @@ public static ClientAuthenticationType parse(String value, String settingKey) {
         }
     }
 
+    public enum TokenType {
+        ID_TOKEN("id_token"),
+        ACCESS_TOKEN("access_token");
+
+        private final String value;
+
+        TokenType(String value) {
+            this.value = value;
+        }
+
+        public String value() {
+            return value;
+        }
+
+        public static TokenType parse(String value, String settingKey) {
```

Comment:
nit: code here is perfectly fine ... but [here ](https://github.com/elastic/elasticsearch/blob/main/libs/dissect/src/main/java/org/elasticsearch/dissect/DissectKey.java#L169) is an alternative using EnumSet in case you like the other style better.

Reply:
It's ok with me either way. I updated it to use the stream API as suggested.